### PR TITLE
Add retry logic around logic IndexedDB connections.

### PR DESCRIPTION
**Pull request by avolkovi**
Similar to https://github.com/firebase/firebase-js-sdk/blob/a10c18f8996fc35942779f5fea5690ae5d102bb0/packages/firestore/src/local/simple_db.ts

### CODE REVIEW

File: packages-exp/auth-exp/src/platform_browser/persistence/indexed_db.ts [hunk 1]
```diff
@@ -166,6 +166,7 @@ function deleteObject(db: IDBDatabase, key: string): Promise<void> {
 export const _POLLING_INTERVAL_MS = 800;
 
 class IndexedDBLocalPersistence implements Persistence {
+  private static readonly TRANSACTION_RETRY_COUNT = 3;
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/platform_browser/persistence/indexed_db.ts:169]: Don't use static variables (per Fei's tree shaking work)

Use this as a file-local `const` instead

File: packages-exp/auth-exp/src/platform_browser/persistence/indexed_db.ts [hunk 2]
```diff
@@ -193,14 +194,34 @@ class IndexedDBLocalPersistence implements Persistence {
...
+        return await op(db);
+      } catch (e) {
+        if (numAttempts++ > IndexedDBLocalPersistence.TRANSACTION_RETRY_COUNT) {
+          throw e;
+        }
+        if (this.db) {
+          this.db.close();
+          this.db = undefined;
+        }
+        // TODO: consider adding exponential backoff
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/platform_browser/persistence/indexed_db.ts:220]: Is it worth adding any sort of delay here?

**Review by samhorlbeck — APPROVED** [review]

File: packages-exp/auth-exp/src/platform_browser/persistence/indexed_db.ts (hunk 2 above)
Comment (avolkovi) [reply packages-exp/auth-exp/src/platform_browser/persistence/indexed_db.ts:220]: question for @schmidt-sebastian I think he mentioned they had this in Firestore but I couldn't find proof

File: packages-exp/auth-exp/src/platform_browser/persistence/indexed_db.ts (hunk 1 above)
Comment (avolkovi) [reply packages-exp/auth-exp/src/platform_browser/persistence/indexed_db.ts]: done

File: packages-exp/auth-exp/src/platform_browser/persistence/indexed_db.ts (hunk 2 above)
Comment (schmidt-sebastian) [reply packages-exp/auth-exp/src/platform_browser/persistence/indexed_db.ts:221]: We retry immediately at this scope, but at the higher level we retry with backoff: https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/util/async_queue.ts#L361

**Review by schmidt-sebastian — APPROVED** [review]

File: packages-exp/auth-exp/src/platform_browser/persistence/indexed_db.ts (hunk 2 above)
Comment (avolkovi) [reply packages-exp/auth-exp/src/platform_browser/persistence/indexed_db.ts:221]: thanks, will consider adding that in a followup if this doesn't fix it

**Pull request merged**